docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md

```diff
@@ -22,71 +24,158 @@ Before starting the migration process, ensure the following:
+      After the migration, select one of the following:
 
-- **Stop replication**: If no further synchronization is needed, and you are ready
-  to switch to Aiven for Valkey after testing.
-- **Keep replicating**: If continuous data synchronization is needed. Avoid system
-  updates or configuration changes during active replication to  prevent unintended
-  migrations.
+      - **Stop replication**: If no further synchronization is needed, and you are ready
+        to switch to Aiven for Valkey after testing.
+      - **Keep replicating**: If continuous data synchronization is needed. Avoid system
+        updates or configuration changes during active replication to  prevent unintended
+        migrations.
 
-:::note
-When replication is active, Aiven for Valkey ensures your data stays in sync by
-continuously synchronizing new writes from the source database.
+      :::note
+      When replication is active, Aiven for Valkey ensures your data stays in sync by
+      continuously synchronizing new writes from the source database.
+      :::
+
+</TabItem>
+<TabItem value="k8" label="Aiven Operator for Kubernetes®">
+:::tip
+See
+[Aiven Operator for Kubernetes configuration options for Valkey](https://aiven.github.io/aiven-operator/api-reference/valkey.html).
+:::
+
+1. Use the Aiven API to change the type of your service from `redis` to `valkey`: Call the
+   [ServiceServiceTypeUpdate](https://api.aiven.io/doc/#tag/Service/operation/ServiceServiceTypeUpdate)
+   API endpoint, replacing `PROJECT_NAME` and `SERVICE_NAME` with meaningful values.
+
+   ```bash {4}
+   curl -X PATCH "https://api.aiven.io/v1/project/PROJECT_NAME/service/SERVICE_NAME/service_type" \
+   -H "Authorization: Bearer $AIVEN_TOKEN" \
+   -H "Content-Type: application/json" \
+   -d '{"service_type": "valkey"}'
+   ```
+
+1. [Get authenticated and authorized](https://aiven.github.io/aiven-operator/authentication.html)
+   to use the Aiven Operator for Kubernetes.
+1. In your service manifest file, change `kind: Redis` to `kind: Valkey`:
+
+   ```yaml {3}
+   # sample_redis.yaml
+   apiVersion: aiven.io/v1alpha1
+   kind: Valkey
+   metadata:
+     name: METADATA_NAME
+   spec:
+     authSecretRef:
+       name: aiven-token
+       key: token
+
+     connInfoSecretTarget:
+       name: SECRET_NAME
+
+     project: PROJECT_NAME
+
+     cloudName: CLOUD_AND_REGION_NAME
+     plan: SERVICE_PLAN_NAME
+
+     maintenanceWindowDow: DAY_OF_WEEK
+     maintenanceWindowTime: HH:MM:SS
+
+     technicalEmails:
+       - email: test@example.com
+
+     connInfoSecretTargetDisabled: false
+   ```
+
+1. Update the service by applying the configuration:
+
+   ```bash
+   kubectl apply -f sample_redis.yaml
+   ```
+
+   This creates an Aiven for Valkey resource: a service managed by the Aiven Operator for
+   Kubernetes.
+
+1. In the Aiven for Redis manifest file, add `controllers.aiven.io/deletion-policy: Orphan`
+   under `metadata` > `annotations`.
```

Comment:
I think this is technically not correct. The user should not update the manifest file, but instead change the manifest (using their k8s managing tool of choice) to update the orphaned resources (the kind: Redis) that was upgraded to Valkey annotations to include `deletion-policy: Orphan`.

@byashimov Are you aware of common patterns people use to upgrade annotations? How should we word this?

Reply:
In that case, how the complete flow should look like?

1. Change the service type via Aiven API.
2. Create a new Valkey resource.
3. Destroy the old Redis resource.

So now changing `kind` and no adding annotations via K8? What am I missing?
@rriski @byashimov